odbc_fdw.c

```diff
@@ -155,9 +182,208 @@ odbc_fdw_handler(PG_FUNCTION_ARGS)
 	fdwroutine->IterateForeignScan = odbcIterateForeignScan;
 	fdwroutine->ReScanForeignScan = odbcReScanForeignScan;
 	fdwroutine->EndForeignScan = odbcEndForeignScan;
+	fdwroutine->ImportForeignSchema = odbcImportForeignSchema;
 	PG_RETURN_POINTER(fdwroutine);
 }
 
+void
+init_odbcFdwOptions(odbcFdwOptions* options)
+{
+	options->dsn          = NULL;
+	options->driver       = NULL;
+	options->host         = NULL;
+	options->port         = NULL;
+	options->database     = NULL;
+	options->username     = NULL;
+	options->password     = NULL;
+	options->schema       = NULL;
+	options->table        = NULL;
+	options->prefix       = NULL;
+	options->sql_query    = NULL;
+	options->sql_count    = NULL;
+	options->mapping_list = NIL;
+}
+
+void
+copy_odbcFdwOptions(odbcFdwOptions* to, odbcFdwOptions* from)
+{
+	if (to && from)
+	{
+		to->dsn          = from->dsn;
+		to->driver       = from->driver;
+		to->host         = from->host;
+		to->port         = from->port;
+		to->database     = from->database;
+		to->username     = from->username;
+		to->password     = from->password;
+		to->schema       = from->schema;
+		to->table        = from->table;
+		to->prefix       = from->prefix;
+		to->sql_query    = from->sql_query;
+		to->sql_count    = from->sql_count;
+		to->mapping_list = from->mapping_list;
+	}
+}
+
+/*
+ * Replace empty string by null pointer
+ */
+static void
+normalize_empty_string(char **str)
+{
+	if (*str && !**str)
+	{
+		*str = NULL;
```

Comment:
if the pointer originally points to a reserved mem buffer, shouldn't this function free it?